src/CityOnPlanet.cpp

```diff
@@ -202,123 +312,345 @@ CityOnPlanet::~CityOnPlanet()
 CityOnPlanet::CityOnPlanet(Planet *planet, SpaceStation *station, const Uint32 seed)
 {
 	// beware, these are not used in this function, but are used in subroutines!
+	m_body = planet->GetSystemBody();
 	m_planet = planet;
 	m_frame = planet->GetFrame();
 	m_detailLevel = Pi::detail.cities;
```

Comment:
we will do it again in `AddStaticGeomsToCollisionSpace`

Reply:
I'm not really a fan of the way that assignment is hidden in AddStaticGeoms, but the whole thing is not the best way to handle it anyways. I'll get rid of the duplicated assign here, it's really not necessary.